File: src/main/java/com/tartner/dancehours/web/home/HomeController.java
package com.tartner.dancehours.web.home;

import com.tartner.dancehours.querymodel.danceuser.DanceUserListQueryModel;
import com.tartner.dancehours.querymodel.danceuser.DanceUserListTO;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;

import java.security.Principal;
import java.util.List;

@Controller
public class HomeController {
	@Autowired private DanceUserListQueryModel queryModel;

	@RequestMapping(value = "/", method = RequestMethod.GET)
	public String index(Principal principal) {
        final List<DanceUserListTO> allUsers = queryModel.getAllUsers();

        return principal != null ? "home/homeSignedIn" : "redirect:/login";
	}
}

File: src/main/java/com/tartner/dancehours/web/home/HomeController.kt
package com.tartner.dancehours.web.home

import com.tartner.dancehours.querymodel.danceuser.DanceUserListQueryModel
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import java.security.Principal

@Controller
public class HomeController {
    @Autowired private val queryModel: DanceUserListQueryModel? = null

    @RequestMapping(value = "/", method = arrayOf(RequestMethod.GET))
    public fun index(principal: Principal?): String {
        val allUsers = queryModel!!.getAllUsers()

        return if (principal != null) "home/homeSignedIn" else "redirect:/login"
    }
}
